### PR TITLE
Support custom delimiter text for multi-values

### PR DESCRIPTION
**Pull request by adamjarling**
## What does this do?
Supports a custom delimiter for multi-value display.  Previously, for example a `metadata` item had a shape like this:

```jsx
{
    label: {
      none: ["Subject"],
    },
    value: {
      none: [
        "Masks",
        "Commedia dell'arte",
        "Italian drama (Comedy)",
        "Pantaloon (Fictitious character)",
      ],
    },
  },
```

The UI display value would be `"Masks, Commedia dell'arte, Italian drama (Comedy), Pantaloon (Fictitious character)"`

Now a user can pass in a custom delimiter value:

```jsx
<Metadata metadata={...} customValueDelimiter="!" />
```
and it will display as: `"Masks! Commedia dell'arte! Italian drama (Comedy)! Pantaloon (Fictitious character)"`

This is especially useful when combined with the `customValueContent` prop, which lets a consuming application target (and style) a group of elements.

This issue also introduces a pattern for wrapping Nectar components with `Context` to pass values directly to deeply nested components.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,33 +29,34 @@
...
   "dependencies": {
-    "@stitches/react": "^1.2.7",
-    "hls.js": "^1.1.5",
-    "react": "^16.8  || ^17.0 || ^18.0",
-    "react-dom": "^16.8  || ^17.0 || ^18.0",
-    "sanitize-html": "^2.7.0"
+    "@stitches/react": "^1.2.8",
+    "hls.js": "^1.2.8",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
```
Comment (mathewjordan) [inline package.json:58]: This is okay, right? We had been using the ORs `"react": "^16.8  || ^17.0 || ^18.0"` but I just want to make sure...

File: src/components/Value/CustomValue.tsx [hunk 2]
```diff
@@ -20,7 +23,7 @@ const CustomValue: React.FC<NectarCustomValue> = ({
   return (
     <StyledCustomValue as={as} lang={lang}>
       {entries?.map((entry, index) => [
-        index > 0 && ", ",
+        index > 0 && `${delimiter} `,
```
Comment (mathewjordan) [inline src/components/Value/CustomValue.tsx:26]: Should this be...

```suggestion
        index > 0 && `${delimiter}`,
```
_OR_
```suggestion
        index > 0 && delimiter,
```

to be even more explicit?

**Review by mathewjordan — COMMENTED** [review]
I really like this, I just left a few comments. What are you thoughts?

File: package.json (hunk 1 above)
Comment (adamjarling) [reply package.json:58]: Ah, good catch.  I'll switch back.

File: src/components/Value/CustomValue.tsx (hunk 2 above)
Comment (adamjarling) [reply src/components/Value/CustomValue.tsx:26]: @mathewjordan Mmmm, not sure.  I did keep the space in here, but you're suggesting the default `delimiter` value would be `", "`?    So if I'm using this, it'd be...

`<Metadata ... customValueDelimiter="*, " />

**Pull request merged**